### PR TITLE
Add subdiv ops

### PR DESCRIPTION
**Pull request by hoshiryu**
* **Please check if the PR fulfills these requirements**
- [x] Is the Pull-Request done against the right branch:
  - `master`: for hotfixes not impacting the API
  - `master-v1`: for changes related to the Radium Stable Release V1.
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add subdivision operations registration and replay.
Add Attribute in TopologicalMesh for the vertex index in the input/output TriangleMesh.

* **What is the current behavior?** (You can also link to an open issue here)
Performing subdivision at each frame (e.g. after skinning) is very slow.

* **What is the new behavior (if this is a feature change)?**
Performing subdivision at each frame takes 30ms on average for 2 subdivision iterations.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
A Plugin for post-process subdivision has been added at `https://gitlab.com/Storm-IRIT/radium-official-plugins/postprocesssubdivplugin.git`.
Currently, when activated, the subdivision is performed at each frame, even if nothing changed (e.g. paused animation, or no animation at all). Moreover, there is no guarantee that the post-process subdivision will be done as the last task due to parallel scheduling of tasks.

### CODE REVIEW

**Comment by nmellado** [comment]
This PR includes a new plugin.
We are still discussing on modalities for official plugin support, we will work on this PR once modalities are fixed.

**Comment by hoshiryu** [comment]
Shall I split the PR into the following?
 - Add subdivision operations registration and replay
 - Add Post Subdiv Plugin

**Comment by hoshiryu** [comment]
`MSBUILD : error MSB4017: The build stopped unexpectedly because of an unexpected logger failure`
Can someone relaunch the windows build please?

**Comment by nmellado** [comment]
Validation of this PR is waiting for the plugin's PR acceptance:
https://gitlab.com/Storm-IRIT/radium-official-plugins/postprocesssubdivplugin/merge_requests/1

**Comment by nmellado** [comment]
Waiting for CRON build to finish, the plugin compilation seem to works well.
https://travis-ci.org/STORM-IRIT/Radium-Engine/builds/449208300

File: src/Core/File/MaterialData.hpp [hunk 1]
```diff
@@ -33,9 +33,9 @@ namespace Asset {
  *      Make your loader instantiate the right "Asset" MaterialData class while loading material
  * data from a file.
  *
- *      The active system (GeometrySystem is the Radium Default), will then automatically use your
- * new material and technique so that the rendering will be fine. When writing your own system, see
- * GeometrySystem implementation as an example.
+ *      The active system (FancyMesh is the Radium Default), will then automatically use your new
+ * material and technique so that the rendering will be fine. When writing your own system, see
+ * FancyMesh implementations as an example.
```
Comment (nmellado) [inline src/Core/File/MaterialData.hpp]: Please revert this file

**Comment by nmellado** [comment]
You have a linking error on MacOSX with OpenMesh (https://github.com/STORM-IRIT/Radium-PluginExample/blob/master/laplacian-smoothing/CMakeLists.txt#L50).
You need to link with openmesh libraries in your plugin (see https://github.com/STORM-IRIT/Radium-PluginExample/blob/master/laplacian-smoothing/CMakeLists.txt#L50)

**Comment by nmellado** [comment]
Great, your plugin has passed the CI tests, so we can accept the PR.

**Pull request merged**